confdgnmi/testtool/gNMI_Interface/GetLibrary.py

```diff
@@ -0,0 +1,139 @@
+            return 0
+        # trace(self.last_response)
+        return sum(len(n.update) for n in self.last_response.notification)
+
+    def supported_models_should_include(self, model_name: str) -> bool:
+        # TODO - rewrite to more efficient any()...
+        models = self.get_supported_model_names()
+        assert model_name in models, f'CapabilityResponse does NOT include \"{model_name}\"'
+
+    def get_supported_model_names(self):
+        """ Return list of all the models supported by device/server.\n
+            This is retrieved from the `CapabilityRequest`'s supported_models property. """
+        response = self._client.get_capabilities()
+        return [model.name for model in response.supported_models]
+
+    def cleanup_getrequest_parameters(self):
+        """ Clear all parameters of following `GetRequest` to be empty. """
+        self.params = GetRequestParameters.new()
+
+    def set_prefix_to(self, prefix: str):
+        """ Set the `prefix` parameter of the next `GetRequest` to specified value. """
+        self.params.prefix = prefix
+
+    def set_encoding_to(self, encoding: str):
+        """ Set the `Encoding` parameter of the next `GetRequest` to specified value. """
+        self.params.encoding = encoding_str_to_int(encoding)
+
+    def set_datatype_to(self, data_type: str):
+        """ Set the `DataType` parameter of the next `GetRequest` to specified value. """
+        self.params.get_type = datatype_str_to_int(data_type, no_error = True)
+
+    def add_path_parameter(self, path: str):
+        """ Add a path parameter into collected array for next `GetRequest`. """
+        params = self.params
+        if params.paths is None:
+            params.paths = []
+        params.paths.append(path)
+
+    def dispatch_get_request(self):
+        """ Dispatch the GetRequest towards server and store the received response.\n
+            Parameters of the request are set according to previously set placeholder values
+            (see `set_..._to()` methods). """
+        self.cleanup_last_request_results()
+        try:
+            kwargs = self.params.to_kwargs(self.default_encoding)
+            self.last_response = self._client.get_public(**kwargs)
+        except Exception as ex:
+            self.last_exception = ex
+            return
+
+    def get_last_flattened_updates(self) -> List[List[UpdatePayload]]:
+        if self.last_response is None:
+            return None
+        notifications = self.last_response.notification
+        updates = []
+        for n in notifications:
+            for update in n.update:
+                trace(update)
+                updates.append(UpdatePayload.from_obj(update))
+        return updates
+
+    def _updates_include(self, text: str) -> bool:
+        updates = self.get_last_flattened_updates()
+        if updates is None:
+            return False
+        trace(updates)
+        return any(text in update.value for update in updates)
```

Comment:
I'm afraid this may not be enough - this says that e.g. `"name"` appears as a string in the textual representation of the latest update.

Reply:
Right, this needs to be made more strict/specific to returned data format.
I should have added TODO/notice for this "draft" implementation.
Will fix before merge...